### ISSUE TITLE
z-index change breaks jQueryUI use

### ISSUE THREAD

**Issue by gaelicmichael** [Leaflet/Leaflet#3690]
I've been using Leaflet 0.7.3 very successfully with jQueryUI. It is clear that the z-index values or logic has changed in 1.beta, as my own divs that were once working are now overlaid by Leaflet and jQueryUI modal dialog boxes are hidden underneath the map pane.

**Comment by mourner** [comment]
cc @patrickarlt

**Comment by mourner** [comment]
Yeah, the z-index values were bumped. Can you set up a JSFiddle to demonstrate the problem?

**Comment by Eschon** [comment]
I don't know about jQueryUI, but I had some divs over my map.
After updating to 1.0 beta 1 the were below the map so I just increased the z-index on my divs and it worked fine.

**Comment by gaelicmichael** [comment]
It's obvious how to modify my own CSS to give things a great z-index, but the issue is compatibility with a large library like jQueryUI, which had no previous problems. It's not reasonable to have to modify the jQueryUI code itself to enable inter-operability, though I haven't looked yet for means of specifying the use of z-index values for modals and other widgets -- I'm going to look this morning.

Regardless, the usage and values of z-index in Leaflet needs to be well documented in order for developers to anticipate such issues.

**Comment by gaelicmichael** [comment]
This entry may help those with this problem: http://stackoverflow.com/questions/16917430/jquery-ui-1-10-dialog-and-zindex-option

**Comment by mourner** [comment]
So, any chance to see a JSFiddle?

**Comment by jaymmartin** [comment]
See fiddle here: https://jsfiddle.net/onwu4kv6/

Seems like by default jQueryUI dialog z-index is 100 and default map pane has a z-index of 100.

**Comment by IvanSanchez** [comment]
This can be worked around very easily by manually specifying a lower `z-index` for the map.

The underlying problem is that the map panes share the same [stacking context](https://developer.mozilla.org/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context) as the siblings of the map container. So it's not the z-index of the map container, it's the z-index of the panes.

I'm partial to add a [`isolation: isolate` CSS rule](https://developer.mozilla.org/en-US/docs/Web/CSS/isolation) to `.leaflet-container`, as this allows the map containers to forcefully have their own [stacking context](https://developer.mozilla.org/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context) without forcing a specific z-index. Or set `z-index:0` altogether so it works on IE also.

@jaymmartin Thanks for the fiddle!

**Comment by jieter** [comment]
@IvanSanchez Why "partial to add `isolation: isolate;` and not just add it? What would be a reason not to?

**Comment by IvanSanchez** [comment]
@jieter Browser support. See https://developer.mozilla.org/en-US/docs/Web/CSS/isolation#Browser_compatibility

**Comment by JorisDebonnet** [comment]
I just encountered a similar problem, with the leaflet map covering my hovering menu bar.

A simple fix could be to just add `.leaflet-container {z-index:0}` to the `leaflet.css` stylesheet. No?

**Comment by Malvoz** [comment]
@JorisDebonnet The difference between `isolation: isolate` and setting `z-index` is that the former doesn't potentially override any custom `z-index` set on `leaflet-container`.

**Comment by derekmil** [comment]
Hello, I'm still have trouble, as I have set z-index to 0 and done isolation: isolate, and it still seems that I need to set the zindex to 400 to get anything to render over the map. Am I correct that you are referring to the .leaflet-container in leaflet.css?

**Comment by patrickarlt** [comment]
@derekmil can you post a minimal reproduction somewhere like CodePen so I can take a look?
